pythran/pythonic/include/numpy/ndarray/tofile.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef PYTHONIC_INCLUDE_NUMPY_NDARRAY_TOFILE_HPP
+#define PYTHONIC_INCLUDE_NUMPY_NDARRAY_TOFILE_HPP
+
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/utils/numpy_conversion.hpp"
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/types/str.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+
+  namespace ndarray
+  {
+    template <class T, class pS>
+    void tofile(types::ndarray<T, pS> const &expr, types::str const &file_name,
+                types::str const &sep = "", types::str const &format = "");
+
+    NUMPY_EXPR_TO_NDARRAY0_DECL(tofile);
```

Comment:
As you're doing a single pass on the data, you don't need that conversion, you could take any array expression and iterate over it!